app/src/main/java/com/zeapo/pwdstore/utils/Extensions.kt

```diff
@@ -188,3 +188,39 @@ val RevCommit.time: Date
         val epochMilliseconds = epochSeconds * 1000
         return Date(epochMilliseconds)
     }
+
+fun FragmentManager.performTransaction(destinationFragment: Fragment, @IdRes containerViewId: Int = android.R.id.content) {
+    this.commit {
+        beginTransaction()
+        setCustomAnimations(
+            R.animator.slide_in_left,
+            R.animator.slide_out_left,
+            R.animator.slide_in_right,
+            R.animator.slide_out_right)
+        replace(containerViewId, destinationFragment)
+    }
+}
+
+fun FragmentManager.performTransactionWithBackStack(destinationFragment: Fragment, @IdRes containerViewId: Int = android.R.id.content) {
+    this.commit {
+        beginTransaction()
+        addToBackStack(destinationFragment.tag)
+        setCustomAnimations(
+            R.animator.slide_in_left,
+            R.animator.slide_out_left,
+            R.animator.slide_in_right,
+            R.animator.slide_out_right)
+        replace(containerViewId, destinationFragment)
+    }
+}
+
+fun FragmentManager.performSharedElementTransaction(destinationFragment: Fragment, views: List<View>, @IdRes containerViewId: Int = android.R.id.content) {
+    this.commit {
+        beginTransaction()
+        for (view in views) {
+            addSharedElement(view, view.transitionName)
+        }
+        addToBackStack(destinationFragment.tag)
+        replace(containerViewId, destinationFragment)
+    }
+}
```

Comment:
Move them to a separate `FragmentExtensions` file, things are getting pretty wild in here. Need to redistribute some more but that can go in a separate PR.